TAScheduler/UnitTests/tests.py

```diff
@@ -10,40 +12,180 @@
+            home_address="123 Street",
+            phone_number=1234567890
+        )
+        self.userObj = UserObj(user_info)
+    def test_get_name(self):
+        self.assertEqual(self.userObj.getName(), "Test User")
+class TestUserGetRole(TestCase): # Alec
+    def setUp(self):
+        admin_user_info = User.objects.create(
+            email_address="admin@example.com",
+            password="adminpass",
+            first_name="Admin",
+            last_name="User",
+            home_address="123 Admin Street",
+            phone_number=1234567890
+        )
+        admin_info = Administrator.objects.create(user=admin_user_info)
+        self.adminObj = AdminObj(admin_info)
+    def test_get_role(self):
+        self.assertEqual(self.adminObj.getRole(), "admin")  # Replace "user" with the actual role name if different
+class TestAdminInit(TestCase): # Alec
+    def setUp(self):
+        admin_user_info = User.objects.create(
+            email_address="admin@example.com",
+            password="adminpass",
+            first_name="Admin",
+            last_name="User",
+            home_address="123 Admin Street",
+            phone_number=1234567890
+        )
+        admin_model = Administrator.objects.create(user=admin_user_info)
+        self.adminObj = AdminObj(admin_model)
+    def test_admin_init(self):
+        self.assertEqual(self.adminObj.user.email_address, "admin@example.com")
+class TestAdminCreateCourse(TestCase): # Alec
+    def setUp(self):
+        admin_user_info = User.objects.create(
+            email_address="admin@example.com",
+            password="adminpass",
+            first_name="Admin",
+            last_name="User",
+            home_address="123 Admin Street",
+            phone_number=1234567890
+        )
+        admin_model = Administrator.objects.create(user=admin_user_info)
+        self.adminObj = AdminObj(admin_model)
+        course_info = {
+            'course_id': 101,
+            'semester': 'Fall 2023',
+            'name': 'Intro to Testing',
+            'description': 'A course about unit testing',
+            'num_of_sections': 3,
+            'modality': 'Online',
+            'credits': 4
+        }
+        self.courseObj = CourseObj(course_info)
+    def test_create_course(self):
+        created_course = self.adminObj.createCourse(self.courseObj)
+        self.assertIsNotNone(created_course)
+        self.assertEqual(created_course.name, 'Intro to Testing')
+class TestAdminCreateUser(TestCase): # Alec
+    def setUp(self):
+        admin_user_info = User.objects.create(
+            email_address="admin@example.com",
+            password="adminpass",
+            first_name="Admin",
+            last_name="User",
+            home_address="123 Admin Street",
+            phone_number=1234567890
+        )
+        admin_model = Administrator.objects.create(user=admin_user_info)
+        self.adminObj = AdminObj(admin_model)
+
+    def test_create_user(self):
```

Comment:
Every time admin creates a user there needs to be a role assigned as well

Reply:
how might I do that? i cant find where it is assigned in the model

Reply:
do we want to add the to the User in models.py?

Reply:
nevermind, got it.

Reply:
I was just saying that user_info should have a line that goes 'role', 'TA' or something like that.

Reply:
i know, but because role isn't part of the views_methods.py, how do we just add that?

Reply:
I imagine that admin.createUser is creating an entry in the user database and linking that to a TA or instructor entry. We in the test are passing a dictionary of everything admin needs to make this happen. But we also need to pass a role entry so that the admin can link the user to a specific role

Reply:
So its specific to this method, gotcha gotcha